### ISSUE TITLE
Bulk handling for suggestions

### ISSUE THREAD

**Issue by paulpach** [WeblateOrg/weblate#671]
A person came to my site and added hundreds of suggestions.

Currently I have to review each suggestion one at a time, and accept them one at a time.

It would be great if there was a "zen mode"  for reviewing large number of suggestions.
##

**Comment by Frexuz** [comment]
Would like to see this too!

**Comment by derikson** [comment]
This looks to me like a duplicate of #849.

**Comment by mark-yougov** [comment]
It would be great if there was a "zen mode" for not only reviewing a large number of suggestions but also bulk approving all the suggestions in the language. ie having an "Approve All Suggestions" button.

**Comment by exzhawk** [comment]
My implementation https://github.com/exzhawk/weblate/commit/ce5106970502e43825340ece0608beaa90b74c84
Known issue: No vote/spam function.
Screenshot
![image](https://user-images.githubusercontent.com/7038938/110248404-da16e200-7fab-11eb-9adf-3a1476fb4ccd.png)

**Comment by comradekingu** [comment]
@exzhawk Avatars is a great idea.
I think it would be better to put them left-aligned (with a mouseover for the name) so that the text has a clear A-B comparison instead of being offset.

**Comment by comradekingu** [comment]
@exzhawk Avatars is a great idea.
I think it would be better to put them left-aligned (with a mouseover for the name) so that the text has a clear A-B comparison instead of being offset.

**Comment by nijel** [comment]
Related: https://github.com/WeblateOrg/weblate/discussions/8554#discussioncomment-4612940

**Comment by juliancoffee** [comment]
Is it something that is accepted as a feature? If it just needs an actual implementation, I could try to create a PR.

**Comment by nijel** [comment]
@juliancoffee Yes, it is accepted that we need such a feature. The exact design was not yet discussed, so please post your proposal here before implementing so that feedback can be provided.

**Comment by juliancoffee** [comment]
Well, I quite like the design in https://github.com/WeblateOrg/weblate/issues/671#issuecomment-841025994.

It seems to include both a "Suggest" button with an exclamation sign, as well as all suggestions with accept/accept and edit/trash buttons. And looks quite clean. Except, I don't think we need to have suggestions to be different colours or smth, having an avatar and all the buttons gives IMO enough information to differentiate between current translation and suggestions.

And maybe add a vote button along with accept/accept and edit/trash.

**Comment by nijel** [comment]
Yes, that makes sense. I think the main question is whether to make it available in the zen mode, or it would be a separate view for dealing with the suggestions. I can see benefits in both approaches. Having it included makes the suggestions available while translating. Having it separate can make the UI more streamlined for suggestions removing not needed parts.

**Comment by juliancoffee** [comment]
> Having it separate can make the UI more streamlined for suggestions removing not needed parts.
I think we can just dispatch on whether the user has translation rights?

**Comment by nijel** [comment]
User having permission to accept suggestions will often have also permission to edit the translation, so that is IMHO not the way to decide this.

**Comment by juliancoffee** [comment]
Yeah, I think having them separate might be both simpler to implement and simpler to use. But we can adjust it afterwards.

**Comment by juliancoffee** [comment]
here's a start #11187, I actually went via always showing suggestions and translations at the same time

**Comment by pastk** [comment]
It'd be great to have this functionality indeed.

Note there is a request to bulk edit (dismiss) automatic suggestions also https://github.com/WeblateOrg/weblate/issues/10396 which had been marked as a duplicate of this issue, but perhaps should be implemented differently.

**Comment by nijel** [comment]
https://github.com/WeblateOrg/weblate/pull/17712 adds option to accept all suggestion from a single user.
